v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/processing/dml/DMLGenerator.java

```diff
@@ -336,31 +333,19 @@ private static String getMappedColumnValue(
           valuesJson.getJSONArray(colName).toList().stream()
               .map(String::valueOf)
               .collect(Collectors.joining(","));
-    } else if ("STRING".equals(colType)
-        && ("binary".equals(sourceColDef.getType().getName())
-            || "varbinary".equals(sourceColDef.getType().getName()))) {
-
-      // Spanner has the hex string in this case
-      try {
-        colInputValue = new String(Hex.decodeHex(valuesJson.getString(colName)));
-      } catch (DecoderException e) {
-        // return the same string value
-        colInputValue = valuesJson.getString(colName);
-      }
```

Comment:
why are we removing this? Shouldn't we have a HEX() on this?

Reply:
Hex handling is handled in the getHexString method. It adds HEX there.